android/uhabits-core/src/main/java/org/isoron/uhabits/core/ui/NotificationTray.java

```diff
@@ -202,23 +202,16 @@ public void onPostExecute()
                 return;
             }
 
-            if (!habit.hasReminder()) {
-                systemTray.log(String.format(
-                        Locale.US,
-                        "Habit %d does not have a reminder. Skipping.",
-                        habit.id));
-                return;
-            }
-
             systemTray.showNotification(habit, getNotificationId(habit), timestamp,
                 reminderTime);
         }
 
         private boolean shouldShowReminderToday()
         {
             if (!habit.hasReminder()) return false;
-            Reminder reminder = habit.getReminder();
+            if (habit.isArchived()) return false;
```

Comment:
Instead of modifying this method, could you add a new block in `onPostExecute` instead, where we print `"habit is archived, skipping"`? I would like the log to tell us **precisely** why have we decided not to show a notification. This helps us debug "notifications are not showing up" bug reports.